docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,465 @@
+      "type": "object",
+      "required": [
+        "group",
+        "kind",
+        "version"
+      ],
+      "properties": {
+        "envVarPrefix": {
+          "type": "string"
+        },
+        "group": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "namespace": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+
+```
+
+Note - removed `description` fields to make the sample concise. 
+</details>
+
+
+
+### User has access to ClusterServiceVersion
+
+The approach is to go through the keys provided by the user against the ones present in the ClusterServiceVersion's CRDDescription. if the user has provided parameters which aren't present in the Description ( SpecDescriptors ) then we show an error with all the parameters that are incorrectly provided.
+
+## "odo catalog describe service"
+
+The json output for all the approaches below would be -
+
+```
+{
+  "kind": ...
+
+  "spec":[
+    {
+      "fieldPath": "zookeeper.resources.limit.min",
+      "displayName": "Limit Min",
+      "Description": "<description>",
+      "type":"string"
+    },
+    ....
+  ],
+}
+
+```
+Note - this is not "openAPIv3schema" format.
+
+### Approach where user has access to the CustomResourceDefinition
+`odo catalog service describe strimzi-cluster-operator.v0.21.1/Kafka -o json` would show a flat converted version of `openAPIv3schema` with same structure as `ClusterServiceVersion` with an extra addition of `type` information being present.
+
+For human readable output -
+```
+- FieldPath: zookeeper.resources.limit.min
+  DisplayName: Limit Min
+  Description: <description>
+  Type: string
+   
+- FieldPath: zookeeper.storage.type
+  Type: string
+```
+
+### Approach where user only has access to ClusterServiceVersion 
+
+`odo catalog service describe strimzi-cluster-operator.v0.21.1/Kafka -o json` would show the `ClusterServiceVersion`'s `CRDDescription` shown in the `Fetch ClusterServiceVersion to generate the information` section
+
+For human readable output a non tablular approach would be used.
+
+```
+- FieldPath: zookeeper.resources
+   DisplayName: Zookeeper Resource Requirements
+   Description: Limits describes the minimum/maximum amount of compute resources required/allowed
+- FieldPath: zookeeper.storage.type
+   DisplayName: ....
+   ...
+
+```
+
+
+
+## "odo service create"
+
+For any scenerio the user would provide the parameters in flat format i.e. `odo service create -p zookeeper.storage.type=ephemeral`. We would build the full parameters map using these flat key values then in case of `openapiv3schema`, validate it against the jsonschema. For `ClusterServiceVersion` we already have the flat path values so we validate them before building the parameters map.
+
+### --input-file flag
```

Comment:
isn't this flag called `--from-file` now? Do you want to rename it?

Reply:
ack? by mistake